### PR TITLE
Use adjustPassManager equivalent for NPM

### PR DESCRIPTION
**Pull request by aankit-ca**
Add call to TargetMachine::registerPassBuilderCallbacks to allow targets to add passes to the pass pipeline using the New Pass Manager
@alinas PTAL

### CODE REVIEW

File: src/CodeGen_LLVM.cpp [hunk 1]
```diff
@@ -1340,6 +1340,9 @@ void CodeGen_LLVM::optimize_module() {
         }
     }
 
+    if (tm)
```
Comment (steven-johnson) [inline src/CodeGen_LLVM.cpp:1343]: Always enclose if stmts in brackets in Halide, even for single-line stmts

Comment (aankit-ca) [reply src/CodeGen_LLVM.cpp]: Sorry. DId not know that. Corrected.

**Review by steven-johnson — COMMENTED** [review]
LGTM

**Comment by steven-johnson** [comment]
clang-tidy complains:

```
/home/runner/work/Halide/Halide/src/CodeGen_LLVM.cpp:1344:13: error: no member named 'registerPassBuilderCallbacks' in 'llvm::TargetMachine' [clang-diagnostic-error]
        tm->registerPassBuilderCallbacks(pb, debug_pass_manager);
```

**Comment by aankit-ca** [comment]
> clang-tidy complains:
> 
> ```
> /home/runner/work/Halide/Halide/src/CodeGen_LLVM.cpp:1344:13: error: no member named 'registerPassBuilderCallbacks' in 'llvm::TargetMachine' [clang-diagnostic-error]
>         tm->registerPassBuilderCallbacks(pb, debug_pass_manager);
> ```

The required LLVM patch was merged today: https://reviews.llvm.org/D88138

https://github.com/llvm/llvm-project/blob/93a1fc2e18b452216be70f534da42f7702adbe1d/llvm/include/llvm/Target/TargetMachine.h#L300

**Review by steven-johnson — APPROVED** [review]
Buildbots finally green. Thanks!

**Comment by alinas** [comment]
Sorry for the delayed reply, this looked good.
Thank you for adding the LLVM support!

**Pull request merged**